Hireling fix 1

Blogaugis opened the pull request:
Added a "Combat Knife" to Skitarii soldiers;
Fixed Inquisition Crusader issue with gear item slot having an item belonging in a weapon group;
Added "Ork Armour" to "Ork Sniper";
Lots of TODO comments added;
Fixes the Crusaders not having Armour

Blogaugis committed:
Update scr_add_man.gml

Added a "Combat Knife" to Skitarri soldiers;
Fixed "Inquisition Crusader" issue with gear item slot having an item belonging in a weapon group;
Added "Ork Armour" to "Ork Sniper"
Lots of TODO comments added.

Blogaugis committed:
Update scr_add_man.gml

Fixes the Crusaders not having Armour

OH296 commented on scripts/scr_add_man/scr_add_man.gml:
```diff
@@ -35,12 +35,63 @@ function scr_add_man(man_role, target_company, spawn_exp, spawn_name, corruption
 		if (other_gear = true) {
 	// Factions 1-5 are part of Imperial family
 		// Faction 1 - Space Marine
+			/* TODO - add the right faction stuff, and move the "switch (man_role) {" before uncommenting
```
we will not be making new additions using the scr_add_man script like this in the furture it's terrible practise, but these can stay commented for now as guidane

Blogaugis replied:
Ok, I'll try fiddling around with the scr_marine_struct in future PRs.

OH296 commented on scripts/scr_add_man/scr_add_man.gml:
```diff
@@ -9,7 +9,7 @@ function scr_add_man(man_role, target_company, spawn_exp, spawn_name, corruption
 	// That should be sufficient to add stuff in a highly modifiable fashion
 
-	var non_marine_roles = ["Skitarii", "Techpriest", "Crusader", "Sister of Battle", "Sister Hospitaler", "Ranger", "Ork Sniper", "Flash Git"];
+	var non_marine_roles = ["Skitarii", "Techpriest", "Crusader", "Sister of Battle", "Sister Hospitaler", "Ranger", "Ork Sniper", "Flash Git"]; // if adding new hirelings, don't forget to update this list
```
this list will be irrelevant soon this is just retrospective of the old code base

Blogaugis replied:
Understood.

OH296 reviewed (changes requested):
I would suggest you familiarise yourself with the concept of OOP and the scr_marine_struct rather than focussing on scr_add_man its a poorly written file that we will probably not continue to expand and support

Blogaugis committed:
Update scr_marine_struct.gml

(Re)introduced stat randomization to hirelings,
Added more TODO and other comments.

Blogaugis committed:
Light Bolter and Power Armour addition

Adding Light Bolter and Light Power armour, fixing the issue of SoBs having ranged weapon penalties.

Blogaugis committed:
Combat Knife addition

Skitarii receive a Combat Knife melee weapon.

EttyKitty reviewed (commented):
I think considering that https://github.com/EttyKitty/ChapterMaster/pull/27 already adds light power armour and light bolter, you may want to merge loadout bits of this PR into that PR instead, and close this one.

Blogaugis commented:
Understood, I'll try to merge it to #27 .

Blogaugis committed:
Merge branch 'pr/14-hiresup2' into pr/14-hiresup1

Blogaugis commented:
Done, it seems.
